### PR TITLE
update messages/oneOf/reference OK from operations

### PR DESCRIPTION
**Pull request by vvilerio**
Adding change/update reference in messages(oneOf) from an operation

Warning : depends on https://github.com/Apicurio/apicurio-data-models/pull/364

update
n

### CODE REVIEW

File: front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/definition/oneof-row.component.ts [hunk 1]
```diff
@@ -87,13 +82,17 @@ export class OneOfRowComponent extends AbstractRowComponent<AaiMessage, Simplifi
         var vals: Array<Value> = [];
         var doc: AaiDocument = (<AaiDocument>this.item.ownerDocument());
         doc.components.getMessagesList().filter(f => f.getName() !== this.deref(item).getName()).forEach(mess => {
-            vals.push(new Value(mess.getName(), "#/component/messages/".concat(mess.getName())));
+            vals.push(new Value(mess.getName(), "#/components/messages/".concat(mess.getName())));
```
Comment (EricWittmann) [inline front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/definition/oneof-row.component.ts:85]: Nice fix here.

**Comment by EricWittmann** [comment]
Overall looks fine, but my proposal in data-models impacts this.

File: front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/channel/operation/payload-tab.component.ts [hunk 2]
```diff
@@ -123,8 +118,14 @@ export class PayloadTabComponent extends AbstractBaseComponent {
...
 
-        let command: ICommand = CommandFactory.createChangePayloadRefCommand_Aai20(nt.type, this.message.parent() as AaiOperation);
-        this.commandService.emit(command);
+        if(this.from == PayloadType.OPE.toString()){
+            let command: ICommand = CommandFactory.createChangePayloadRefCommand_Aai20(nt.type, this.message.parent() as AaiOperation);
+            this.commandService.emit(command);
+        }else if(this.from == PayloadType.COMP.toString()){
+            let command: ICommand = CommandFactory.createChangePayloadRefCommand_Aai20(nt.type, this.message.parent() as AaiComponents, this.message._name); //the name like key
+            this.commandService.emit(command);
+        }
```
[the remark this answers is not in the document]
Comment (vvilerio) [reply front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/channel/operation/payload-tab.component.ts:127]: OK, You can modify them, when you'll have merged data-models

[the remark this answers is not in the document]
Comment (vvilerio) [reply front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/channel/operation/payload-tab.component.ts:127]: I tested with data-models version 1.1.20 

image


and I get the following error

image

do you have this problem ?

[the remark this answers is not in the document]
Comment (EricWittmann) [reply front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/channel/operation/payload-tab.component.ts:127]: Trying this now.  Will report back.

[the remark this answers is not in the document]
Comment (EricWittmann) [reply front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/channel/operation/payload-tab.component.ts:127]: This doesn't happen for me.  I've manually merged this PR and fixed some stuff up.  I'm going to merge to `master` and you can give it a try.

**Comment by EricWittmann** [comment]
Merged manually with changes.  There are a lot of bugs in the AsyncAPI editor that I encountered while working on this.  I don't have bandwidth to work on it right now - but I do think we need to keep this feature in Beta state for now.

File: front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/channel/operation/payload-tab.component.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (vvilerio) [reply front-end/studio/src/app/pages/apis/{apiId}/editor/_components/forms/channel/operation/payload-tab.component.ts:127]: best wishes to you, I'm back from vacations, could you remind me if it is "createChangePayloadRefCommand_Aai20" that should be replaced by "ChangePropertyCommand"?

**Comment by lerio29** [comment]
Hello Eric,

some works about messages are not in the last release, could you publish one ?

**Comment by vvilerio** [comment]
Hello Eric,

some works about messages are not in the last release, could you publish one ?

**Comment by EricWittmann** [comment]
Sure thing!

**Comment by vvilerio** [comment]
@EricWittmann Thanks

**Pull request merged**